src/orders-v2/order-refunds/order-refunds-view-model.ts

```diff
@@ -0,0 +1,176 @@
+  TransactionActionEnum,
+  TransactionEventFragment,
+  TransactionEventTypeEnum,
+} from "@dashboard/graphql";
+
+export type OrderRefundDisplay = {
+  id: string;
+  type: "standard" | "manual";
+  status: OrderGrantedRefundStatusEnum;
+  amount: {
+    amount: number;
+    currency: string;
+  };
+  reason: string | null;
+  createdAt: string;
+  user: {
+    email: string;
+    firstName: string;
+    lastName: string;
+  } | null;
+};
+
+export type OrderRefundState =
+  | "noTransactionsToRefund"
+  | "allTransactionsNonRefundable"
+  | "refundable";
+
+export abstract class OrderRefundsViewModel {
+  static getRefundState(transactions: OrderDetailsFragment["transactions"]): OrderRefundState {
+    if (transactions.length === 0) {
+      return "noTransactionsToRefund";
+    }
+
+    if (
+      transactions.every(transaction => !transaction.actions.includes(TransactionActionEnum.REFUND))
+    ) {
+      return "allTransactionsNonRefundable";
+    }
+
+    return "refundable";
+  }
+
+  static canEditRefund(refund: OrderRefundDisplay): boolean {
+    const isSuccessful = refund.status === OrderGrantedRefundStatusEnum.SUCCESS;
+    const isPending = refund.status === OrderGrantedRefundStatusEnum.PENDING;
+    const isManual = refund.type === "manual";
+
+    // Can only edit refunds that are not successful, not pending, and not manual
+    return !isSuccessful && !isPending && !isManual;
+  }
+
+  private static groupEventsByPspReference = (
+    events: TransactionEventFragment[],
+  ): Record<string, TransactionEventFragment[]> => {
+    return events?.reduce<Record<string, TransactionEventFragment[]>>((acc, event) => {
+      if (!acc[event.pspReference]) {
+        acc[event.pspReference] = [];
+      }
+
+      acc[event.pspReference].push(event);
+
+      return acc;
+    }, {});
+  };
+
+  private static findLatestEventWithUserAuthor = (
+    eventGroup: TransactionEventFragment[],
+  ): TransactionEventFragment | null => {
+    return eventGroup.find(event => event.createdBy?.__typename === "User") || null;
+  };
+
+  private static mapEventToRefundStatus = (
+    event: TransactionEventFragment,
+  ): OrderGrantedRefundStatusEnum => {
+    switch (event.type) {
+      case TransactionEventTypeEnum.REFUND_SUCCESS:
+        return OrderGrantedRefundStatusEnum.SUCCESS;
+      case TransactionEventTypeEnum.REFUND_FAILURE:
+        return OrderGrantedRefundStatusEnum.FAILURE;
+      case TransactionEventTypeEnum.REFUND_REQUEST:
+        return OrderGrantedRefundStatusEnum.PENDING;
+      default:
+        return OrderGrantedRefundStatusEnum.NONE;
+    }
+  };
+
+  private static determineCreatorDisplay = (
+    creator: AppAvatarFragment | StaffMemberAvatarFragment | null,
+  ): OrderRefundDisplay["user"] => {
+    if (creator?.__typename === "User") {
+      return {
+        email: creator.email,
+        firstName: creator.firstName,
+        lastName: creator.lastName,
+      };
+    }
+
+    return null;
```

Comment:
we don't render app?

Reply:
not sure when its null. Its automatic, created by sAelor? if so, maybe we should show saleor logo or something similar

Reply:
yes - I copy / pasted it from previous implementation - I will change that in the next PR